src/graph/csr.rs

```diff
@@ -50,46 +50,46 @@ impl Default for CsrLayout {
 /// neighbor list is defined by `offsets[u + 1] - offsets[u]`. The neighbor list
 /// of `u` is defined by the slice `&targets[offsets[u]..offsets[u + 1]]`.
 #[derive(Debug)]
-pub struct Csr<Node: Idx> {
-    offsets: Box<[Node]>,
-    targets: Box<[Node]>,
+pub struct Csr<Index: Idx, T> {
+    offsets: Box<[Index]>,
+    targets: Box<[T]>,
 }
 
-impl<Node: Idx> Csr<Node> {
-    pub(crate) fn new(offsets: Box<[Node]>, targets: Box<[Node]>) -> Self {
+impl<Index: Idx, T> Csr<Index, T> {
```

Comment:
Could use a default type to signal the common case, e.g.

```suggestion
impl<Index: Idx, T> Csr<Index, T = Index> {
```

which would allow using `Csr<Node>` as an alias for `Csr<Node, Node>`.

Reply:
So nice! :heart: :crab: